bpo-37942: Improve argument clinic float converter

rhettinger opened the pull request:
<!-- issue-number: [bpo-37942](https://bugs.python.org/issue37942) -->
https://bugs.python.org/issue37942
<!-- /issue-number -->

rhettinger committed:
bpo-37942: Improve argument clinic float converter

serhiy-storchaka reviewed (approved):
LGTM.

Do you want to add the same optimization in `getargs.c`?

rhettinger committed:
Add fast path for exact floats.

Baseline timing:
./python.exe -m timeit -r11 -s 'from statistics import _normal_dist_inv_cdf as f' -s 't=0.75,0.0,1.0' 'f(*t)'
5000000 loops, best of 11: 61.1 nsec per loop

Improved timing:
./python.exe -m timeit -r11 -s 'from statistics import _normal_dist_inv_cdf as f' -s 't=0.75,0.0,1.0' 'f(*t)'
5000000 loops, best of 11: 52.5 nsec per loop

rhettinger committed:
Apply the same fix to getargs.c

rhettinger committed:
Get the casts in the right place.

rhettinger committed:
Update clinic.test

rhettinger committed:
Regenerate the argument clinic files
